crates/matrix-sdk/src/client/builder.rs

```diff
@@ -84,25 +84,14 @@ impl ClientBuilder {
 
     /// Set the homeserver URL to use.
     ///
-    /// This method is mutually exclusive with [`user_id()`][Self::user_id], if
-    /// you set both whatever was set last will be used.
+    /// This method is mutually exclusive with
+    /// [`server_name()`][Self::server_name], if you set both whatever was
+    /// set last will be used.
     pub fn homeserver_url(mut self, url: impl AsRef<str>) -> Self {
         self.homeserver_cfg = Some(HomeserverConfig::Url(url.as_ref().to_owned()));
         self
     }
 
-    /// Set the user ID to discover the homeserver from.
-    ///
-    /// `builder.user_id(id)` is a shortcut for
-    /// `builder.server_name(id.server_name())`.
-    ///
-    /// This method is mutually exclusive with
-    /// [`homeserver_url()`][Self::homeserver_url], if you set both whatever was
-    /// set last will be used.
-    pub fn user_id(self, user_id: &UserId) -> Self {
```

Comment:
Given that `user_id` naming was confusing (sets server, not user), and `server_name_from_user_id` was too long, it seems we could simply just rely on `server_name(user_id.server_name())`

Reply:
Can we please not do this in this PR? I'd rather deprecate it for now, not remove it entirely.